### ISSUE TITLE
Android crash when trying to sign in through Facebook when Firebase is initialised manually

### ISSUE THREAD

**Issue by part22** [firebase/quickstart-unity#461]
## Please fill in the following fields:
Unity editor version: 2017.4.32f
Firebase Unity SDK version: 6.4.0
Firebase plugins in use: Auth
Additional SDKs you are using: Facebook v7.17.2
Platform you are using the Unity editor on: Windows
Platform you are targeting: Android

## Please describe the issue here:
On device, the app crashes when trying to sign in a user with Facebook auth provider.
**The crash only occurs when manually initialising Firebase.**

**Pre-conditions**
- Firebase app configured to accept facebook sign in
- Facebook app configured to let firebase handle OAuth

**Steps**
1. Launch the app
2. Sign in with Facebook
3. Wait to be logged in
4. Try to sign in with Firebase using Facebook
5. **BOOM**

In the editor, following these steps works perfectly fine. I can see the user has been added to the users in the firebase app with the right provider.

Logcat output:
> 09-07 10:49:56.528  5165  5195 E firebase: app != nullptr
> 09-07 10:49:56.530  5165  5195 E CRASH   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 00000000
> 09-07 10:49:56.530  5165  5195 E CRASH   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***

I am **not** using _Firebase.FirebaseApp.DefaultInstance_ instead I am initialising Firebase myself as follows:
```csharp
[SerializeField]
private TextAsset _firebaseConfigFile;

private Firebase.Auth.FirebaseAuth CreateAuth()
{
    var firebaseOptions = Firebase.AppOptions.LoadFromJsonConfig(_firebaseConfigFile.text);
    var mainFirebaseApp = Firebase.FirebaseApp.Create(firebaseOptions, "main_app");
    return Firebase.Auth.FirebaseAuth.GetAuth(mainFirebaseApp);
}
```
__firebaseConfigFile_ is assigned the json config file downloaded from my Firebase app.

**Note**
On device, I can signin/signup a user with email and password without a problem.

## Please answer the following, if applicable:
Yes, I have been able to reproduce the issue with the test app.

**Comment by markchandler-google** [comment]
Hi part22,

I think the issue your hitting is that FirebaseApp is getting disposed due to being in a temp variable and thus the auth object is in an invalid state. Keep the app object alive and see if that fixes your issue.

**Comment by part22** [comment]
Hi @markchandler-google,

Thanks for your answer.
I have already tried somehting similar (see code below) for the same reason you have mentioned. Unfortunately, I have had the same problem.

In that scenario the code looked like:
``` csharp
[SerializeField]
private TextAsset _firebaseConfigFile;

private Firebase.FirebaseApp _mainFirebaseApp;
private Firebase.Auth.FirebaseAuth _auth;

private void InitAuth()
{
    var firebaseOptions = Firebase.AppOptions.LoadFromJsonConfig(_firebaseConfigFile.text);
    _mainFirebaseApp = Firebase.FirebaseApp.Create(firebaseOptions, "main_app");
    _auth = Firebase.Auth.FirebaseAuth.GetAuth(mainFirebaseApp);
}
```

**Comment by markchandler-google** [comment]
Since you can repo this with the sample app are you able to create a gist of the code and config so I can reproduce the issue?

**Comment by markchandler-google** [comment]
With help from @stewartmiles looks like that for android the default instance of the app must always exist due to logic grabbing the Jni Environment pointer: https://github.com/firebase/firebase-cpp-sdk/blob/master/auth/src/android/credential_android.cc#L315

I will put a bug in for this but for now you can fix the crash by creating the default instance of app and holding onto it in your code.

**Comment by a-maurice** [comment]
Hi all,

This issue should be resolved in the new version of the Unity SDK, 6.6.0, available at https://firebase.google.com/download/unity

Thanks for bringing this issue to our attention!
